### PR TITLE
Set a custom nodeselector for scratch builds if low_priority_node_selector config param is set

### PR DESCRIPTION
**Pull request by vrutkovs**
Signed-off-by: Vadim Rutkovsky <vrutkovs@redhat.com>

### CODE REVIEW

**Comment by rcerven** [comment]
how does that test tests anything? you flexmock build_request and then you update json

**Comment by vrutkovs** [comment]
The tests are flexmocking `BuildRequest`'s string, received from configuration and checking that parsed dict is found in a json, used to create a new Openshift `Build` object.

The only thing missing is direct read from configuration, but it looks trivial enough to me (and somehow already being covered by other config tests)

File: osbs/api.py [hunk 1]
```diff
@@ -294,6 +294,13 @@ def _create_scratch_build(self, build_request):
         build_json['spec']['serviceAccount'] = 'builder'
         build_json['metadata']['labels']['scratch'] = 'true'
 
+        if build_request.low_priority_node_selector:
+            nodeselector = {}
+            for item in build_request.low_priority_node_selector.split(','):
```
Comment (lcarva) [inline osbs/api.py]: Can't we do the string manipulation in only in conf.py? and assume that at the api level it should already be a list?

Comment (vrutkovs) [inline osbs/api.py]: I guess we could, yes, I don't really have a preference

Comment (twaugh) [inline osbs/api.py]: My preference would be to do this in conf.py, so that at the API level it is a dict.

**Comment by vrutkovs** [comment]
Updated the PR: nodeselector now being parsed in `conf.py`, tests are reading the configuration explicitely

File: osbs/conf.py [hunk 2]
```diff
@@ -491,3 +491,14 @@ def get_can_orchestrate(self):
...
                                "info_url_format")
+
+    def get_low_priority_node_selector(self):
+        nodeselector = {}
+        nodeselector_str = self._get_value("low_priority_node_selector", self.conf_section,
+                                           "low_priority_node_selector")
+        if nodeselector_str:
+            for item in nodeselector_str.split(','):
+                splitted_list = [x.strip() for x in item.strip().split('=')]
+                nodeselector[splitted_list[0]] = splitted_list[1]
```
Comment (twaugh) [inline osbs/conf.py]: I don't like the variable name `splitted_list`. How about this?:

```
constraints = nodeselector_str.split(',')
nodeselector = dict([constraint.split('=', 1) for constraint in constraints])
```

Comment (vrutkovs) [inline osbs/conf.py]: Nice, although it doesn't strip the values. I'll rework it a bit

Comment (vrutkovs) [inline osbs/conf.py]: Reworked it have a (rather complex) list comprehension statement

File: osbs/conf.py [hunk 3]
```diff
@@ -491,3 +491,13 @@ def get_can_orchestrate(self):
...
         return self._get_value("info_url_format", self.conf_section,
                                "info_url_format")
+
+    def get_low_priority_node_selector(self):
+        nodeselector = {}
+        nodeselector_str = self._get_value("low_priority_node_selector", self.conf_section,
+                                           "low_priority_node_selector")
+        if nodeselector_str:
+            nodeselector = dict([[x.strip() for x in constraint.split('=', 1)]
+                                 for constraint in nodeselector_str.split(',')])
```
Comment (twaugh) [inline osbs/conf.py]: My preference would be to keep to only one `for` in each comprehension, but otherwise this looks fine.

Comment (vrutkovs) [inline osbs/conf.py]: Rewrote this to use two dict comprehension - first to split keys and values - and the second one to strip extra spaces

**Pull request merged**